main.c

```diff
@@ -0,0 +1,124 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+struct Node {
+  int data;
+  struct Node* nextEntry;
+  struct Node* prevEntry;
+};
+
+struct Node* head;
+
+struct Node* CreateNode (int var)
+{
+  struct Node* newNode = (struct Node*)malloc(sizeof(struct Node));
+  newNode->data = var;
+  newNode->prevEntry = NULL;
+  newNode->nextEntry = NULL;
+  return newNode;
+}
+
+void insertAtBegin (int var)
+{
+  struct Node* temp = CreateNode(var);
+  // Replacing head Node with new one
+  if (head) {
+    temp->nextEntry = head;
+    head->prevEntry = temp;
+  } 
+  head = temp;
+}
+
+void insertAtEnd (int var)
+{
+  struct Node* temp = CreateNode(var);
+  if (head) {
+    //Getting the last ptr in list
+    struct Node* lastEntryPtr = head;
+    while(lastEntryPtr->nextEntry) {
+      lastEntryPtr = lastEntryPtr->nextEntry;
+    }
+    lastEntryPtr->nextEntry = temp;
+    temp->prevEntry = lastEntryPtr;
+  } else {
+    head = temp;
+  }
+}
+
+void deleteNodeAtBegin ()
+{
+  if (head) {
+    struct Node* nextInRow = head->nextEntry;
+    free(head);
+    if (nextInRow) {
+      head = nextInRow;
+      head->prevEntry = NULL;
+    } else {
+      head = NULL;
+    }
+  } else {
+    printf("Nothing do delete - list is empty\n");
+  }
+}
+
+void deleteNodeAtEnd ()
+{
+  if (head) {
+    //Getting the last ptr in list
+    struct Node* lastEntryPtr = head;
+    while(lastEntryPtr->nextEntry) {
+      lastEntryPtr = lastEntryPtr->nextEntry;
+    }
+    struct Node* prevFromLastOne = lastEntryPtr->prevEntry;
+    free(lastEntryPtr);
+    if (prevFromLastOne) {
+      prevFromLastOne->nextEntry = NULL;
+    } else {
+      head = NULL;
+    }
+  } else {
+    printf("Nothing do delete - list is empty\n");
+  }
+}
+
+void printList ()
+{
+  struct Node* iterPtr = head;
+  while(iterPtr) {
+    printf("%d\n", iterPtr->data);
+    iterPtr = iterPtr->nextEntry;
+  }
+  return;
+}
+
+void printListReverse ()
+{
+  // Looking for last node in list
+  struct Node* iterPtr = head;
+  while(iterPtr->nextEntry) {
+    iterPtr = iterPtr->nextEntry;
+  }
+  // Printing in reverse
+  while(iterPtr) {
+    printf("%d\n", iterPtr->data);
+    iterPtr = iterPtr->prevEntry;
+  }
+}
+
+int main ()
+{
+  insertAtBegin(0);
```

Comment:
Please make a reusable list, so in ```main()``` you could create however many lists you'd like.